python/tank/authentication/login_dialog.py

```diff
@@ -148,18 +189,28 @@ def __init__(self, is_session_renewal, hostname=None, login=None, fixed_host=Fal
         self.ui._2fa_code.editingFinished.connect(self._strip_whitespaces)
         self.ui.backup_code.editingFinished.connect(self._strip_whitespaces)
 
-        self._update_ui_according_to_sso()
+        self._query_task = QuerySiteAndUpdateUITask(self)
+        self._query_task.finished.connect(self._toggle_sso)
+        self._update_ui_according_to_sso_support()
+
+        # We want to wait until we know if the site uses SSO or not, to avoid
+        # flickering GUI.
+        self._query_task.wait()
```

Comment:
What if this check takes a long time? The user won't be able to hit cancel and quit.